platforms/chibios/drivers/wear_leveling/wear_leveling_efl.c

```diff
@@ -14,26 +14,36 @@ static flash_sector_t first_sector = WEAR_LEVELING_EFL_FIRST_SECTOR;
 static flash_sector_t first_sector = UINT16_MAX;
 #endif // defined(WEAR_LEVELING_EFL_FIRST_SECTOR)
 
-static flash_sector_t sector_count = UINT16_MAX;
-static BaseFlash *    flash;
-
-static volatile bool is_issuing_read    = false;
-static volatile bool ecc_error_occurred = false;
+#if !defined(WEAR_LEVELING_EFL_OMIT_LAST_SECTOR_COUNT)
+#    define WEAR_LEVELING_EFL_OMIT_LAST_SECTOR_COUNT 0
+#endif // WEAR_LEVELING_EFL_OMIT_LAST_SECTOR_COUNT
 
 // "Automatic" detection of the flash size -- ideally ChibiOS would have this already, but alas, it doesn't.
-static inline uint32_t detect_flash_size(void) {
-#if defined(WEAR_LEVELING_EFL_FLASH_SIZE)
-    return WEAR_LEVELING_EFL_FLASH_SIZE;
-#elif defined(FLASH_BANK_SIZE)
-    return FLASH_BANK_SIZE;
-#elif defined(FLASH_SIZE)
-    return FLASH_SIZE;
-#elif defined(FLASHSIZE_BASE)
+#if (!defined(WEAR_LEVELING_EFL_FLASH_SIZE) && defined(FLASH_BANK_SIZE))
+#    define WEAR_LEVELING_EFL_FLASH_SIZE FLASH_BANK_SIZE
+#elif (!defined(WEAR_LEVELING_EFL_FLASH_SIZE) && defined(FLASH_SIZE))
+#    define WEAR_LEVELING_EFL_FLASH_SIZE FLASH_SIZE
+#elif (!defined(WEAR_LEVELING_EFL_FLASH_SIZE) && defined(FLASHSIZE_BASE))
 #    if defined(QMK_MCU_SERIES_STM32F0XX) || defined(QMK_MCU_SERIES_STM32F1XX) || defined(QMK_MCU_SERIES_STM32F3XX) || defined(QMK_MCU_SERIES_STM32F4XX) || defined(QMK_MCU_SERIES_STM32G4XX) || defined(QMK_MCU_SERIES_STM32L0XX) || defined(QMK_MCU_SERIES_STM32L4XX) || defined(QMK_MCU_SERIES_GD32VF103)
-    return ((*(uint32_t *)FLASHSIZE_BASE) & 0xFFFFU) << 10U; // this register has the flash size in kB, so we convert it to bytes
+#        define WEAR_LEVELING_EFL_FLASH_SIZE ((*(uint32_t *)FLASHSIZE_BASE) & 0xFFFFU) << 10U // this register has the flash size in kB, so we convert it to bytes
 #    elif defined(QMK_MCU_SERIES_STM32L1XX)
 #        error This MCU family has an uncommon flash size register definition and has not been implemented. Perhaps try using the true EEPROM on the MCU instead?
 #    endif
+#endif
+
+#if WEAR_LEVELING_EFL_OMIT_LAST_SECTOR_COUNT > WEAR_LEVELING_EFL_FLASH_SIZE
```

Comment:
This check now fails with an expression syntax error if `WEAR_LEVELING_EFL_FLASH_SIZE` is defined to be calculated from the `FLASHSIZE_BASE` register.

Also the check is wrong anyway, because `WEAR_LEVELING_EFL_OMIT_LAST_SECTOR_COUNT` is defined in terms of flash sectors, not kilobytes (and in fact some MCUs have non-uniform flash sector sizes).

Reply:
since we can't avoid doing things in runtime ( at least on STM32), I updated it to run on init and just count sectors